noxfile.py

```diff
@@ -67,48 +67,50 @@ def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> No
             success_codes=[0, NO_TESTS_COLLECTED_EXIT_CODE],
         )
 
-def merge_pytest_benchmark_results(bencher_json_file_name: str) -> None:
-    """Gather all benchmark results files and merge them into a single file."""
-    merged_results = {"benchmarks": []}
-    files = glob.glob("results_*.json")
-    for file in files:
+def valid_results_file(file_name: str) -> bool:
```

Comment:
Why is this needed? It seems currently it is not an error if the file is e.g. corrupted.

Reply:
This is needed because some of the sub packages do not have any benchmark tests, but they generate empty json files. For this reason, we need this function to check these cases.

Reply:
Indeed this is what I still prefer. Ideally, I would like to delete the subpackage parametrization and run pytest-benchmarks in the model folder. In this way, we will upload way less stuff in Bencher cloud. Regarding your question, by merging the different json files, I had some failures to push the reports to Bencher (weirdly enough only for the self-hosted version). This is way I proceeded with the subpackage by subpackage approach.

Although, let me know what you think about the  subpackage parametrization and I could change it.

Reply:
@havogt
FYI, I have applied the change, where I remove the subpackages. I prefer way more this version. It is simpler and very straightforward. Shall we proceed like this?